### ISSUE TITLE
Charging Station > Enhance firmware upgrade tab to reduce misusage

### ISSUE THREAD

**Issue by LucasBrazi06** [LucasBrazi06/ev-dashboard#2126]
One of our customer managed to use this feature to upload the firmware and endup in a blocked situation.

Add this new capability in the template and hide/show the corresponding tab in the UI and enforce a backend check:

![image.png](https://images.zenhubusercontent.com/5dfcf8cef5388bdc792007df/ef890e8a-5c9f-4c57-bd76-c4621a65a353)

Add modifications to:
+ permit to reset the firmware status in DB from the dashboard; 
+ document or enhance the firmware upgrade tab usage: 
	+ list firmware files in DB
	+ permit to chose one
	+ permit to choose between HTTP or HTTPS for firmware download with a default to HTTP

**Comment by jerome-benoit** [comment]
In that specific case, remote firmware upgrade is supported by the charging station. 

The customer in question have used a feature without knowing how to use it and made a mistake. The fix here is to add an action allowing to reset the firmware upgrade status in the DB to cope with firmware upgrade processus failure or misusage. 

In the long run, we also have to deal with charging station feature enablement based on the OCPP key SupportedFeatureProfiles CSL parsing when there's no template available yet or when the feature is not listed in the template. And also control the template features value with that key to trap template errors.

**Comment by LucasBrazi06** [comment]
No, again for the nth time, we have separated the capabilities for this reason: we enable them only when tested on our side.
We just follow what we have done so far.
Please proceed.

**Comment by jerome-benoit** [comment]
> No, again for the nth time, we have separated the capabilities for this reason: we enable them only when tested on our side.

For the same number of times, it's not a viable solution that already expose a large number of downside: 
+ No automated handling of the number of phases (and the workaround so far require manual changes directly in the DB); 
+ No way to fully test an charging station for the first time:
	+ Template can be done only after the first connection; 
	+ Done template need to wait for the next deployment. 
To test all the features, we have to make several meetings for no good reason and it's hard to explain to vendors. It make loose time to everybody.  

Propositions to deal properly with supported charging stations and unsupported ones have already been discussed at length and accepted to be on the development roadmap since months, minutes here: https://github.com/LucasBrazi06/ev-server/issues/1759 

And we are now waiting.  

> We just follow what we have done so far.

When things done do not match real world usage, they have to be changed.

---

To come back to that specific issue: 

+ OCPP firmware upgrade is supported by the charging station; 
+ OCPP firmware upgrade is tested and works. 

The issue is that in case of failure, the firmware update status cannot be reset is DB from the dashboard. Adding a feature flag for firmware upgrade in template will not solve anything here and is already planned: https://github.com/LucasBrazi06/ev-dashboard/issues/1755

The solution here is:
+ to permit to reset the firmware status in DB from the dashboard; 
+ to document or enhance the firmware upgrade tab usage: 
	+ list firmware files in DB
	+ permit to chose one
	+ permit to choose between HTTP or HTTPS for firmware download with a default to HTTP

**Comment by LucasBrazi06** [comment]
> * No automated handling of the number of phases (and the workaround so far require manual changes directly in the DB);

Nothing to do with the current issue

> No way to fully test an charging station for the first time:
> * Template can be done only after the first connection;
> * Done template need to wait for the next deployment.
>   To test all the features, we have to make several meetings for no good reason and it's hard to explain to vendors. It make loose time to everybody.

This is QA testing, the template should be delivered only for QA

> Propositions to deal properly with supported charging stations and unsupported ones have already been discussed at length and accepted to be on the development roadmap since months, minutes here: [LucasBrazi06/ev-server#1759](https://github.com/LucasBrazi06/ev-server/issues/1759)

Must be summarized in a new issue, cannot go through the whole history again.

> When things done do not match real world usage, they have to be changed.

Agree but this is not the case.

**Comment by LucasBrazi06** [comment]
@jerome-benoit And stop writing lengthy responses and come to me to discuss each point, it'll be more efficient as you already know!!!

**Comment by jerome-benoit** [comment]
> @jerome-benoit And stop writing lengthy responses and come to me to discuss each point, it'll be more efficient as you already know!!!

If from the points developed here and over the previous meetings or exchange and acted to be developed in approved meeting minutes, understanding that: 
+ to test a feature you need to validate, you first need to create the file that will say that the feature can be used and supported (circular dependency, extra unneeded configuration time,  ...) 
+ charging station customer user experience must never be reduced even without official support, just properly stated to him.
+ (ay least ten more argumented reasons if needed)  
is a serious mismatching of real world usage that needs to be solved by following the solution already decided months ago, I do not see how another meeting where they will be repeated once more will change anything. The implementation must just be done now. Period.

**Comment by jerome-benoit** [comment]
Otherwise it's a dupe of https://github.com/LucasBrazi06/ev-dashboard/issues/1755

**Comment by jerome-benoit** [comment]
The enhacements here are really needed, for example on the chargex tenant, they have tried the OCPP fimware update, have failed to use it properly at first and now have a download failed on a charging with no way to reset it themselfves. 

I think that issue can stay open with a normal priority.

**Comment by LucasBrazi06** [comment]
This feature should have not been enabled at first.
Next if chargeX wants the firmware update this is a dev request + impl + test and after it is enabled in the template.
On the test phase we can enable it in the QA system only.

**Comment by jerome-benoit** [comment]
> This feature should have not been enabled at first.
> Next if chargeX wants the firmware update this is a dev request + impl + test and after it is enabled in the template.
> On the test phase we can enable it in the QA system only.

All the charging stations used on all production tenants have: 
+ OCPP firmware support;
+ Tested and validated OCPP firmware support by us. 

So that specific capabilities support would just have show exactly the same and have lead to exactly the same troubles that that issue try to solve. 

And ChargeX people are vendor partner that fully knows how to do firmnware upgrade via OCPP on their own charging stations. They do not need nor want to go through a validation process to use their own station features with a software they pay to supervise them. 
With the current dashoard IHM, they have failed to used the firmware upgrade feature.

**Comment by LucasBrazi06** [comment]
Firmware update has to be handled by us, firmwares are stored in our DB tested with our backend with no regression...
Again we do not allow CPO to update firmware WE have not tested. This is the goal of the templates.
With this in mind, so far we support nothing expect just Schneider at best as far as I know.
We can discuss this today.
